spec/models/saved_claim/caregivers_assistance_claim_spec.rb

```diff
@@ -95,6 +95,141 @@
     end
   end
 
+  describe 'validations' do
+    let(:claim) { build(:caregivers_assistance_claim) }
+
+    before do
+      allow(Flipper).to receive(:enabled?).and_call_original
+    end
+
+    context 'caregiver_retry_form_validation disabled' do
+      before do
+        allow(Flipper).to receive(:enabled?).with(:caregiver_retry_form_validation).and_return(false)
+      end
+
+      context 'no validation errors' do
+        before do
+          allow(JSON::Validator).to receive(:fully_validate).and_return([])
+        end
+
+        it 'returns true' do
+          expect(claim.validate).to eq true
+        end
+      end
+
+      context 'validation errors' do
+        it 'calls the parent method when the toggle is off' do
+          allow(claim).to receive(:form_matches_schema).and_call_original
+
+          claim.validate
+
+          expect(claim).to have_received(:form_matches_schema)
+        end
+      end
+    end
+
+    context 'caregiver_retry_form_validation enabled' do
+      before do
+        allow(Flipper).to receive(:enabled?).with(:caregiver_retry_form_validation).and_return(true)
+      end
+
+      context 'no validation errors' do
+        before do
+          allow(JSON::Validator).to receive(:fully_validate).and_return([])
+        end
+
+        it 'returns true' do
+          expect(Rails.logger).not_to receive(:info)
+            .with('Form validation succeeded on attempt 1/3')
+
+          expect(claim.validate).to eq true
+        end
+      end
+
+      context 'validation errors' do
+        let(:schema_errors) { [{ fragment: 'error' }] }
+
+        context 'when JSON:Validator.fully_validate returns errors' do
+          before do
+            allow(JSON::Validator).to receive(:fully_validate).and_return(schema_errors)
+          end
+
+          it 'adds validation errors to the form' do
+            expect(JSON::Validator).not_to receive(:fully_validate_schema)
```

Comment:
🤔 I'm generally a little skeptical of `expect....not_to` assertions, since this could pass if e.g., `fully_validate_schema` is being invoked but gets renamed?

Reply:
Yeah that's valid. In this case it's a function from a gem we are using, so I'm less worried about it changing? I was mostly trying to validate the difference between this logic and the default `SavedClaim` method logic.